### PR TITLE
Manual page for Meta-Test-Family

### PR DESCRIPTION
**Pull request by phracek**
Signed-off-by: Petr "Stone" Hracek <phracek@redhat.com>

This Pull Request brings all man pages for `mtf`, `mtf-env-clean` and `mtf-env-setup`.

The Review is important.

### CODE REVIEW

File: man/mtf.1 [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+.\"
+.\" This page is distributed under GPL.
+.\"
+.TH mtf 1 2017-11-01 "" "Linux User's Manual"
+.SH NAME
+mtf \- Meta-Test-Family tests container images and modules with user defined tests written in Python and/or linters
+provided by meta-test-family package.
+
+.SH SYNOPSIS
+\fBmtf runs tests for container images
```
Comment (TomasTomecek) [inline man/mtf.1]: should be in the "name" section (or description), see `man cp` for example

File: man/mtf-generator.1 [hunk 2]
```diff
@@ -0,0 +1,16 @@
+.\" Copyright Petr Hracek, 2017
+.\"
+.\" This page is distributed under GPL.
+.\"
+.TH mtf-generator 1 2017-11-01 "" "Linux User's Manual"
+.SH NAME
+mtf-generator \- Meta-Test-Family generates tests written in \fBconfig.yaml\fP file.
+
+.SH SYNOPSIS
+\fBmtf-generator generates tests written \fBconfig.yaml\fP file for using by \fBmtf\fP binary.
```
Comment (TomasTomecek) [inline man/mtf-generator.1]: once again, synopsis should not contain a description

File: man/mtf-env-set.1 [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+.SH SYNOPSIS
+\fBmtf-env-set prepares environment for testing containers
+
+.SH DESCRIPTION
+\fBmtf-env-set\fP is the binary file of Meta-Test-Family. It installs tests dependencies, docker service
+ and starts docker service.
+
+.SH NOTES
+\fBmtf-env-set\fP is useful for users who don't want to care about environment and their settings.
```
Comment (TomasTomecek) [inline man/mtf-env-set.1]: this is a bit harsh, how about "mtf-env-set is useful for people who don't want to set the environment manually and wish to use this executable to do the job"

File: man/mtf-env-clean.1 [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+mtf-env-set \- Part of Meta-Test-Family package. It cleans environment for testing containers.
+
+.SH SYNOPSIS
+\fBmtf-env-clean cleans environment for testing containers
+
+.SH DESCRIPTION
+\fBmtf-env-clean\fP is the binary file for cleaning environment of Meta-Test-Family. It stops docker service.
+
+.SH NOTES
+\fBmtf-env-clean\fP is useful for users who don't want to care about environment and their clean up.
```
Comment (TomasTomecek) [inline man/mtf-env-clean.1]: see comment below

File: man/mtf-env-clean.1 [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+.SH DESCRIPTION
+.PP
+\fBmtf-env-clean\fP is the binary file for cleaning environment of Meta-Test-Family.
+
+.PP
+\fIMODULE=docker\/\fR stops docker service.
+
+.PP
+\fIMODULE=rpm\/\fR does not do cleanup as it can destroy this machine.
```
Comment (bkabrda) [inline man/mtf-env-clean.1]: Rather `does not do any cleanup, as that could potentially uninstall essential packages and therefore damage this machine.`

File: man/mtf-env-clean.1 [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+\fIMODULE=nspawn\/\fR
+.B mtf-env-clean
+
+.SH DESCRIPTION
+.PP
+\fBmtf-env-clean\fP is the binary file for cleaning environment of Meta-Test-Family.
+
+.PP
+\fIMODULE=docker\/\fR stops docker service.
```
Comment (bkabrda) [inline man/mtf-env-clean.1]: Question: does it stop the service itself or just the containers?

File: man/mtf-env-clean.1 [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+.PP
+\fIMODULE=rpm\/\fR does not do cleanup as it can destroy this machine.
+
+.PP
+\fIMODULE=nspawn\/\fR switches back SELinux to original state.
```
Comment (bkabrda) [inline man/mtf-env-clean.1]: ...`switches SELinux back`...

File: man/mtf-env-clean.1 [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+\fIMODULE=rpm\/\fR
+.B mtf-env-clean
+
+\fIMODULE=nspawn\/\fR
+.B mtf-env-clean
+
+.SH DESCRIPTION
+.PP
+\fBmtf-env-clean\fP is the binary file for cleaning environment of Meta-Test-Family.
```
Comment (bkabrda) [inline man/mtf-env-clean.1]: `is a binary file used for cleaning environment after usage of Meta-Test-Family.`

File: man/mtf-env-set.1 [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+\fIMODULE=rpm\/\fR
+.B mtf-env-set
+
+\fIMODULE=nspawn\/\fR
+.B mtf-env-set
+
+
+.SH DESCRIPTION
+.PP
+\fBmtf-env-set\fP is the binary file of Meta-Test-Family package.
```
Comment (bkabrda) [inline man/mtf-env-set.1]: `is a binary file included in the Meta-Test-Family package.`

File: man/mtf-generator.1 [hunk 10]
```diff
@@ -0,0 +1,19 @@
+.\" Copyright Petr Hracek, 2017
+.\"
+.\" This page is distributed under GPL.
+.\"
+.TH mtf-generator 1 2017-11-01 "" "Linux User's Manual"
+.SH NAME
+mtf-generator \- generates tests written in \fBconfig.yaml\fP file.
```
Comment (bkabrda) [inline man/mtf-generator.1]: `generates code for tests written in config.yaml file.`

File: man/mtf-generator.1 [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+
+.SH SYNOPSIS
+.B
+mtf
+
+.SH DESCRIPTION
+\fBmtf-generator\fP is the binary file of Meta-Test-Family package.
```
Comment (bkabrda) [inline man/mtf-generator.1]: `is a binary file included in Meta-Test-Family package.`

File: man/mtf-generator.1 [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+It generates tests written \fBconfig.yaml\fP file for using by \fBmtf\fP binary.
```
Comment (bkabrda) [inline man/mtf-generator.1]: `generates code for tests written in config.yaml file for usage by mtf binary.`

File: man/mtf.1 [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+mtf \- runs tests for container images
+
+.SH SYNOPSIS
+[\fI\,VARIABLES\/\fT]
+.B
+mtf
+[\fI\,OPTIONS\/\fR] PYTHON_TESTS
+
+.SH DESCRIPTION
+\fBmtf\fP is the main binary file of Meta-Test-Family. It tests container images and modules with user defined tests written in Python and/or linters
```
Comment (bkabrda) [inline man/mtf.1]: `container images and/or modules`

File: man/mtf.1 [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+provided by meta-test-family package.
+It runs PYTHON_TESTS in avocado framework.
```
Comment (bkabrda) [inline man/mtf.1]: `using avocado framework as test runner`

File: man/mtf-env-clean.1 (hunk 6 above)
Comment (phracek) [reply man/mtf-env-clean.1]: It stops service;)

**Comment by TomasTomecek** [comment]
Well done!

**Pull request merged**